test/unit/spec/nbextensions/appCell2/widgets/appCellWidget-spec.js

```diff
@@ -1,152 +1,585 @@
+                await this.appCellWidgetInstance.init();
+                await this.appCellWidgetInstance.attach(this.kbaseNode);
+                await this.appCellWidgetInstance.start();
+                // checks the app spec is valid
+                await this.appCellWidgetInstance.run();
+
+                // app spec should have been checked
+                // FSM should have changed state
+                expect(this.appCellWidgetInstance.__fsm().getCurrentState().state).not.toEqual({
+                    mode: 'new',
+                });
+                // UI should have been rendered; only one action button visible
+                const actionButtons = this.kbaseNode.querySelectorAll(`.kb-rcp__action-button`);
+                const hiddenButtons = this.kbaseNode.querySelectorAll(
+                    `.kb-rcp__action-button.hidden`
+                );
+                expect(actionButtons.length).toEqual(hiddenButtons.length + 1);
             });
         });
 
-        xit('has a method stop which returns a Promise', () => {
-            return mockAppCell
-                .init()
-                .then(() => {
-                    return mockAppCell.stop();
-                })
-                .then(() => {
-                    // something to see if it worked.
+        describe('cell stop', () => {
+            beforeEach(async function () {
+                cellStartUp(this);
+                await this.appCellWidgetInstance.init();
+                await this.appCellWidgetInstance.attach(this.kbaseNode);
+                await this.appCellWidgetInstance.start();
+                await this.appCellWidgetInstance.run();
+            });
+
+            it('stops, removing event listeners', async function () {
+                expect(this.appCellWidgetInstance.__fsm().getCurrentState().state).not.toEqual({
+                    mode: 'new',
                 });
+                spyOn(this.appCellWidgetInstance.busEventManager, 'removeAll');
+                await this.appCellWidgetInstance.stop();
+                expect(this.appCellWidgetInstance.busEventManager.removeAll).toHaveBeenCalled();
+            });
+
+            it('detaches from the DOM', async function () {
+                await this.appCellWidgetInstance.stop();
+                expect(this.kbaseNode.childNodes.length).toEqual(1);
+                await this.appCellWidgetInstance.detach();
+                expect(this.kbaseNode.childNodes.length).toEqual(0);
+            });
         });
 
-        xit('has a method detach which returns a Promise', () => {
-            return mockAppCell
-                .init()
-                .then(() => {
-                    return mockAppCell.stop();
-                })
-                .then(() => {
-                    return mockAppCell.detach();
-                })
-                .then(() => {
-                    //see if it worked.
+        describe('the running cell', () => {
```

Comment:
Some tests of the run status messages, since they will be affected by the change